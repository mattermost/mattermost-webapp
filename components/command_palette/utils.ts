// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {Channel} from '@mattermost/types/channels';
import {Team} from '@mattermost/types/teams';

// See LICENSE.txt for license information.
import {CommandPaletteItem} from './command_palette_list_item/command_palette_list_item';
import {
    CmdPalettePictographIcon,
    CmdPalettePictographType,
} from './command_palette_list_item/command_palette_list_item_pictograph';
import {ChannelType, CommandPaletteEntities} from './types';

const channelTypeOrder = {
    [ChannelType.OPEN_CHANNEL]: 0,
    [ChannelType.PRIVATE_CHANNEL]: 1,
    [ChannelType.DM_CHANNEL]: 2,
    [ChannelType.GM_CHANNEL]: 3,
};

function filterName(name: string): string {
    return name.replace(/[.,'"\/#!$%\^&\*;:{}=\-_`~()]/g, ''); // eslint-disable-line no-useless-escape
}

export function channelToCommandPaletteItemTransformer(channels: Channel[], teams: Record<string, Team>): CommandPaletteItem[] {
    return channels.map((channel) => {
        const isArchived = channel.delete_at ? channel.delete_at !== 0 : false;

        let channelIcon = CmdPalettePictographIcon.GLOBE;
        if (isArchived) {
            channelIcon = CmdPalettePictographIcon.ARCHIVE;
        }
        return {
            description: channel.type === ChannelType.DM_CHANNEL ? channel.name : '',
            id: channel.id,
            isDeactivated: isArchived,
            teamName: teams[channel.team_id]?.name,
            pictograph: {
                type: CmdPalettePictographType.ICON,
                pictographItem: channelIcon,
            },
            subType: channel.type as ChannelType,
            title: channel.display_name,
            type: CommandPaletteEntities.Channel,
            isArchived,
        };
    });
}

function sortChannelsByTypeAndDisplayName(locale: string, cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem): number {
    if (channelTypeOrder[cmdPaletteItemA.subType] !== channelTypeOrder[cmdPaletteItemB.subType]) {
        if (channelTypeOrder[cmdPaletteItemA.subType] < channelTypeOrder[cmdPaletteItemB.subType]) {
            return -1;
        }

        return 1;
    }

    const aDisplayName = filterName(cmdPaletteItemA.title);
    const bDisplayName = filterName(cmdPaletteItemB.title);

    if (aDisplayName !== bDisplayName) {
        return aDisplayName.toLowerCase().localeCompare(bDisplayName.toLowerCase(), locale, {numeric: true});
    }

    return cmdPaletteItemA.title.toLowerCase().localeCompare(cmdPaletteItemA.title.toLowerCase(), locale, {numeric: true});
}

function sortChannelsByRecencyAndTypeAndDisplayName(cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem) {
    if (cmdPaletteItemA.lastViewedAt && cmdPaletteItemB.lastViewedAt) {
        return cmdPaletteItemB.lastViewedAt - cmdPaletteItemA.lastViewedAt;
    } else if (cmdPaletteItemA.lastViewedAt) {
        return -1;
    } else if (cmdPaletteItemB.lastViewedAt) {
        return 1;
    }

    // MM-12677 When this is migrated this needs to be fixed to pull the user's locale
    return sortChannelsByTypeAndDisplayName('en', cmdPaletteItemA, cmdPaletteItemB);
}

export function commandPaletteSorter(prefix: string, cmdPaletteItemA: CommandPaletteItem, cmdPaletteItemB: CommandPaletteItem) {
    if (cmdPaletteItemA.isArchived && !cmdPaletteItemB.isArchived) {
        return 1;
    } else if (!cmdPaletteItemA.isArchived && cmdPaletteItemB.isArchived) {
        return -1;
    }

    if (cmdPaletteItemA.isDeactivated && !cmdPaletteItemB.isDeactivated) {
        return 1;
    } else if (cmdPaletteItemB.isDeactivated && !cmdPaletteItemA.isDeactivated) {
        return -1;
    }

    let aDisplayName = cmdPaletteItemA.title.toLowerCase();
    let bDisplayName = cmdPaletteItemA.title.toLowerCase();

    if (cmdPaletteItemA.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemA.subType === ChannelType.DM_CHANNEL &&
        aDisplayName.startsWith('@')) {
        aDisplayName = aDisplayName.substring(1);
    }

    if (cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemB.subType === ChannelType.DM_CHANNEL && bDisplayName.startsWith('@')) {
        bDisplayName = bDisplayName.substring(1);
    }

    const aStartsWith = aDisplayName.startsWith(prefix) || cmdPaletteItemA.description.toLowerCase().startsWith(prefix);
    const bStartsWith = bDisplayName.startsWith(prefix) || cmdPaletteItemB.description.toLowerCase().startsWith(prefix);

    // Open channels user haven't interacted should be at the  bottom of the list
    if (cmdPaletteItemA.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemA.subType === ChannelType.OPEN_CHANNEL &&
        !cmdPaletteItemA.lastViewedAt &&
        ((cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
            cmdPaletteItemB.subType !== ChannelType.OPEN_CHANNEL) || cmdPaletteItemB.lastViewedAt)) {
        return 1;
    } else if ((cmdPaletteItemB.type === CommandPaletteEntities.Channel &&
        cmdPaletteItemB.subType === ChannelType.OPEN_CHANNEL) || !cmdPaletteItemB.lastViewedAt) {
        return -1;
    }

    // Sort channels starting with the search term first
    if (aStartsWith && !bStartsWith) {
        return -1;
    } else if (!aStartsWith && bStartsWith) {
        return 1;
    }
    return sortChannelsByRecencyAndTypeAndDisplayName(cmdPaletteItemA, cmdPaletteItemB);
}
