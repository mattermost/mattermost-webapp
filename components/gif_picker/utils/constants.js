// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    ItemTapAction: {
        OPEN_EMBED_PAGE: 1,
        SHARE: 2,
    },

    Tab: {
        TRENDING: 0,
        REACTIONS: 1,
    },

    appName: {
        mattermost: 'mattermost',
    },
};
