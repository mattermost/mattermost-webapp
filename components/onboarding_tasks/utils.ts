// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const generateTelemetryTag = (category: string, name: string, suffix: string) => {
    return `${category}--${name}--${suffix}`;
};

export function getAnalyticsCategory(isAdmin: boolean) {
    return isAdmin ? 'admin_onboarding' : 'end_user_onboarding';
}
