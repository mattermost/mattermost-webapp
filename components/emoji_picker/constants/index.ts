// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {EmojiCategory} from '@mattermost/types/emojis';

import * as Emoji from 'utils/emoji';

import {Category, Categories} from '../types';

export const RECENT = 'recent';
export const SEARCH_RESULTS = 'searchResults';
export const SMILEY_EMOTION = 'smileys-emotion';
export const CUSTOM = 'custom';

const categoryClass: Map<EmojiCategory, string> = new Map([
    [RECENT, 'icon-clock-outline'],
    [SMILEY_EMOTION, 'icon-emoticon-happy-outline'],
    ['people-body', 'icon-account-outline'],
    ['animals-nature', 'icon-leaf-outline'],
    ['food-drink', 'icon-food-apple'],
    ['activities', 'icon-basketball'],
    ['travel-places', 'icon-airplane-variant'],
    ['objects', 'icon-lightbulb-outline'],
    ['symbols', 'icon-heart-outline'],
    ['flags', 'icon-flag-outline'],
    [CUSTOM, 'icon-emoticon-custom-outline'],
    [SEARCH_RESULTS, ''],
]);

function createCategory(name: EmojiCategory): Category {
    return {
        name,
        id: Emoji.CategoryTranslations.get(name) || '',
        className: categoryClass.get(name) || '',
        message: Emoji.CategoryMessage.get(name)!,
    };
}

export const RECENT_EMOJI_CATEGORY: Pick<Categories, 'recent'> = {recent: createCategory(RECENT)};
export const SEARCH_EMOJI_CATEGORY: Pick<Categories, typeof SEARCH_RESULTS> = {searchResults: createCategory(SEARCH_RESULTS)};

export const CATEGORIES: Categories = Emoji.CategoryNames.
    filter((category) => !(category === 'recent' || category === 'searchResults')).
    reduce((previousCategory, currentCategory) => {
        return {
            ...previousCategory,
            [currentCategory]: createCategory(currentCategory as EmojiCategory),
        };
    }, {} as Categories);

export const EMOJI_PER_ROW = 9; // needs to match variable `$emoji-per-row` in _variables.scss
export const ITEM_HEIGHT = 36; //as per .emoji-picker__item height in _emoticons.scss
export const EMOJI_CONTAINER_HEIGHT = 290; // If this changes, the spaceRequiredAbove and spaceRequiredBelow props passed to the EmojiPickerOverlay must be updated

export const CATEGORY_HEADER_ROW = 'categoryHeaderRow';
export const EMOJIS_ROW = 'emojisRow';

export const EMOJI_SCROLL_THROTTLE_DELAY = 150;
export const EMOJI_ROWS_OVERSCAN_COUNT = 1;

export const CUSTOM_EMOJIS_PER_PAGE = 200;
export const CUSTOM_EMOJI_SEARCH_THROTTLE_TIME_MS = 1000;
