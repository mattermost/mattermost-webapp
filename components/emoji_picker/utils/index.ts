// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Emoji, EmojiCategory} from 'mattermost-redux/types/emojis';

import {compareEmojis} from 'utils/emoji_utils';

import {Categories} from '../types';

function sortEmojis(emojis: Emoji[], recentEmojiStrings: string[], filter: string) {
    const recentEmojis: Emoji[] = [];
    const emojisMinusRecent: Emoji[] = [];

    Object.values(emojis).forEach((emoji) => {
        let emojiArray = emojisMinusRecent;
        const alias = 'short_names' in emoji ? emoji.short_names : [emoji.name];
        for (let i = 0; i < alias.length; i++) {
            if (recentEmojiStrings.includes(alias[i].toLowerCase())) {
                emojiArray = recentEmojis;
            }
        }

        emojiArray.push(emoji);
    });

    return [
        ...recentEmojis.sort((firstEmoji, secondEmoji) => compareEmojis(firstEmoji, secondEmoji, filter)),
        ...emojisMinusRecent.sort((firstEmoji, secondEmoji) => compareEmojis(firstEmoji, secondEmoji, filter)),
    ];
}

function getFilteredEmojis(allEmojis: Record<string, Emoji>, filter: string, recentEmojisString: string[]): Emoji[] {
    const emojis = Object.values(allEmojis).filter((emoji) => {
        const alias = 'short_names' in emoji ? emoji.short_names : [emoji.name];
        for (let i = 0; i < alias.length; i++) {
            if (alias[i].toLowerCase().includes(filter)) {
                return true;
            }
        }

        return false;
    });

    return sortEmojis(emojis, recentEmojisString, filter);
}

export function getEmojisByCategory(allEmojis: Record<string, Emoji>, categories: Categories, categoryName: EmojiCategory, filter: string, recentEmojisString: string[]): Emoji[] {
    if (filter.length) {
        return getFilteredEmojis(allEmojis, filter, recentEmojisString);
    }

    // if (categoryName === 'recent') {
    //     const recentEmojiIds = this.state.categories?.recent?.emojiIds ?? [];
    //     if (recentEmojiIds.length === 0) {
    //         return [];
    //     }

    //     const recentEmojis = new Map();

    //     recentEmojiIds.forEach((emojiId) => {
    //         const emoji = this.state.allEmojis[emojiId];
    //         const emojiSkin = getSkin(emoji);

    //         if (emojiSkin) {
    //             const emojiWithUserSkin = this.convertEmojiToUserSkinTone(emoji, emojiSkin, this.props.userSkinTone);
    //             recentEmojis.set(emojiWithUserSkin.unified, emojiWithUserSkin);
    //         } else {
    //             recentEmojis.set(emojiId, emoji);
    //         }
    //     });

    //     return Array.from(recentEmojis.values());
    // }

    return categories[categoryName].emojiIds.map((emojiId) => allEmojis[emojiId]);
}
