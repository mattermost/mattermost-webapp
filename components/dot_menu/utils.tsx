// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {trackEvent} from 'actions/telemetry_actions';

import {EventTypes, TELEMETRY_CATEGORIES} from 'utils/constants';

export type ChangeEvent = React.KeyboardEvent | React.MouseEvent;

export const trackDotMenuEvent = (e: ChangeEvent, suffix: string): void => {
    if (e.type === EventTypes.CLICK) {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.CLICK + '_' + suffix);
    } else {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.SHORTCUT + '_ ' + suffix);
    }
};
