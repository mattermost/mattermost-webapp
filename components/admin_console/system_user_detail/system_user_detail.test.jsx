// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {Constants} from 'utils/constants';
import SystemUserDetail from 'components/admin_console/system_user_detail/system_user_detail.jsx';

jest.mock('actions/admin_actions.jsx');

describe('components/admin_console/system_user_detail', () => {
    const defaultProps = {
        user: {
            username: 'jim.halpert',
            first_name: 'Jim',
            last_name: 'Halpert',
            nickname: 'Big Tuna',
            id: '1234',
            roles: 'system_user',
        },
        actions: {
            updateUserActive: jest.fn(),
            setNavigationBlocked: jest.fn(),
            addUserToTeam: jest.fn(),
        },
    };

    test('should match default snapshot', () => {
        const props = defaultProps;
        const wrapper = shallow(<SystemUserDetail {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should redirect if user id is not defined', () => {
        const props = {
            ...defaultProps,
            user: {
                id: null,
            },
        };
        const wrapper = shallow(<SystemUserDetail {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot if user is inactive', () => {
        const props = {
            ...defaultProps,
            user: {
                ...defaultProps.user,
                delete_at: 1561683854166,
            },
        };
        const wrapper = shallow(<SystemUserDetail {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot if MFA is enabled', () => {
        const props = {
            ...defaultProps,
            user: {
                ...defaultProps.user,
                mfa_active: 'MFA',
            },
            mfaEnabled: true,
        };
        const wrapper = shallow(<SystemUserDetail {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot if no nickname is defined', () => {
        const props = {
            ...defaultProps,
            user: {
                ...defaultProps.user,
                nickname: null,
            },
        };
        const wrapper = shallow(<SystemUserDetail {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot if LDAP Authentication', () => {
        const props = {
            ...defaultProps,
            user: {
                ...defaultProps.user,
                auth_service: Constants.LDAP_SERVICE,
            },
        };
        const wrapper = shallow(<SystemUserDetail {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot if SAML Authentication', () => {
        const props = {
            ...defaultProps,
            user: {
                ...defaultProps.user,
                auth_service: Constants.SAML_SERVICE,
            },
            mfaEnabled: true,
        };
        const wrapper = shallow(<SystemUserDetail {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});