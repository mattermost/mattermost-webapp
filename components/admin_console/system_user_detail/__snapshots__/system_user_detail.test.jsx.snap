// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/system_user_detail should match default snapshot 1`] = `
<div
  className="SystemUserDetail wrapper--fixed"
>
  <div
    className="admin-console__header with-back"
  >
    <div>
      <Connect(BlockableLink)
        className="fa fa-angle-left back"
        to="/admin_console/user_management/users"
      />
      <FormattedMessage
        defaultMessage="User Configuration"
        id="admin.systemUserDetail.title"
        values={Object {}}
      />
    </div>
  </div>
  <div
    className="admin-console__wrapper"
  >
    <div
      className="admin-console__content"
    >
      <AdminUserCard
        body={
          <React.Fragment>
            <span
              className="SystemUserDetail__position"
            />
            <span
              className="SystemUserDetail__field-label"
            >
              Email
            </span>
            <input
              className="SystemUserDetail__input form-control"
              onChange={[Function]}
              type="text"
            />
            <span
              className="SystemUserDetail__field-label"
            >
              Username
            </span>
            <p>
              jim.halpert
            </p>
            <span
              className="SystemUserDetail__field-label"
            >
              Authentication Method
            </span>
            <p>
              Email
            </p>
            <span
              className="SystemUserDetail__field-label"
            >
              Role
            </span>
            <p>
              <FormattedMessage
                defaultMessage="Member"
                id="admin.user_item.member"
                values={Object {}}
              />
            </p>
          </React.Fragment>
        }
        footer={
          <React.Fragment>
            <AdminButtonOutline
              className="admin-btn-default"
              onClick={[Function]}
            >
              Reset Password
            </AdminButtonOutline>
            <AdminButtonOutline
              className="admin-btn-default"
              onClick={[Function]}
            >
              Deactivate
            </AdminButtonOutline>
          </React.Fragment>
        }
        user={
          Object {
            "first_name": "Jim",
            "id": "1234",
            "last_name": "Halpert",
            "nickname": "Big Tuna",
            "roles": "system_user",
            "username": "jim.halpert",
          }
        }
      />
      <AdminPanel
        className=""
        subtitleDefault="Teams to which this user belongs"
        subtitleId="admin.userManagement.userDetail.teamsSubtitle"
        titleDefault="Team Membership"
        titleId="admin.userManagement.userDetail.teamsTitle"
      >
        <Connect(TeamList)
          userId="1234"
        />
      </AdminPanel>
    </div>
  </div>
  <div
    className="admin-console-save"
  >
    <SaveButton
      btnClass="btn-primary"
      disabled={true}
      extraClasses=""
      onClick={[Function]}
      saving={false}
      savingMessage="Saving Config..."
    />
    <div
      className="error-message"
    >
      <FormError
        error={null}
        errors={Array []}
      />
    </div>
    <Overlay
      animation={[Function]}
      delayShow={400}
      placement="top"
      rootClose={false}
      show={false}
    >
      <Tooltip
        bsClass="tooltip"
        id="error-tooltip"
        placement="right"
      />
    </Overlay>
  </div>
  <Connect(ResetPasswordModal)
    onModalDismissed={[Function]}
    onModalSubmit={[Function]}
    show={false}
    user={
      Object {
        "first_name": "Jim",
        "id": "1234",
        "last_name": "Halpert",
        "nickname": "Big Tuna",
        "roles": "system_user",
        "username": "jim.halpert",
      }
    }
  />
  <ConfirmModal
    confirmButtonClass="btn btn-danger"
    confirmButtonText={
      <FormattedMessage
        defaultMessage="Deactivate"
        id="deactivate_member_modal.deactivate"
        values={Object {}}
      />
    }
    message={
      <div>
        <FormattedMessage
          defaultMessage="This action deactivates {username}. They will be logged out and not have access to any teams or channels on this system. Are you sure you want to deactivate {username}?"
          id="deactivate_member_modal.desc"
          values={
            Object {
              "username": "jim.halpert",
            }
          }
        />
        <strong>
          <br />
          <br />
          <FormattedMessage
            defaultMessage="You must also deactivate this user in the SSO provider or they will be reactivated on next login or sync."
            id="deactivate_member_modal.sso_warning"
            values={Object {}}
          />
        </strong>
      </div>
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <FormattedMessage
        defaultMessage="Deactivate {username}"
        id="deactivate_member_modal.title"
        values={
          Object {
            "username": "jim.halpert",
          }
        }
      />
    }
  />
</div>
`;

exports[`components/admin_console/system_user_detail should match snapshot if MFA is enabled 1`] = `
<div
  className="SystemUserDetail wrapper--fixed"
>
  <div
    className="admin-console__header with-back"
  >
    <div>
      <Connect(BlockableLink)
        className="fa fa-angle-left back"
        to="/admin_console/user_management/users"
      />
      <FormattedMessage
        defaultMessage="User Configuration"
        id="admin.systemUserDetail.title"
        values={Object {}}
      />
    </div>
  </div>
  <div
    className="admin-console__wrapper"
  >
    <div
      className="admin-console__content"
    >
      <AdminUserCard
        body={
          <React.Fragment>
            <span
              className="SystemUserDetail__position"
            />
            <span
              className="SystemUserDetail__field-label"
            >
              Email
            </span>
            <input
              className="SystemUserDetail__input form-control"
              onChange={[Function]}
              type="text"
            />
            <span
              className="SystemUserDetail__field-label"
            >
              Username
            </span>
            <p>
              jim.halpert
            </p>
            <span
              className="SystemUserDetail__field-label"
            >
              Authentication Method
            </span>
            <p>
              MFA
            </p>
            <span
              className="SystemUserDetail__field-label"
            >
              Role
            </span>
            <p>
              <FormattedMessage
                defaultMessage="Member"
                id="admin.user_item.member"
                values={Object {}}
              />
            </p>
          </React.Fragment>
        }
        footer={
          <React.Fragment>
            <AdminButtonOutline
              className="admin-btn-default"
              onClick={[Function]}
            >
              Reset Password
            </AdminButtonOutline>
            <AdminButtonOutline
              className="admin-btn-default"
              onClick={[Function]}
            >
              Deactivate
            </AdminButtonOutline>
            <AdminButtonOutline
              className="admin-btn-default"
              onClick={[Function]}
            >
              Remove MFA
            </AdminButtonOutline>
          </React.Fragment>
        }
        user={
          Object {
            "first_name": "Jim",
            "id": "1234",
            "last_name": "Halpert",
            "mfa_active": "MFA",
            "nickname": "Big Tuna",
            "roles": "system_user",
            "username": "jim.halpert",
          }
        }
      />
      <AdminPanel
        className=""
        subtitleDefault="Teams to which this user belongs"
        subtitleId="admin.userManagement.userDetail.teamsSubtitle"
        titleDefault="Team Membership"
        titleId="admin.userManagement.userDetail.teamsTitle"
      >
        <Connect(TeamList)
          userId="1234"
        />
      </AdminPanel>
    </div>
  </div>
  <div
    className="admin-console-save"
  >
    <SaveButton
      btnClass="btn-primary"
      disabled={true}
      extraClasses=""
      onClick={[Function]}
      saving={false}
      savingMessage="Saving Config..."
    />
    <div
      className="error-message"
    >
      <FormError
        error={null}
        errors={Array []}
      />
    </div>
    <Overlay
      animation={[Function]}
      delayShow={400}
      placement="top"
      rootClose={false}
      show={false}
    >
      <Tooltip
        bsClass="tooltip"
        id="error-tooltip"
        placement="right"
      />
    </Overlay>
  </div>
  <Connect(ResetPasswordModal)
    onModalDismissed={[Function]}
    onModalSubmit={[Function]}
    show={false}
    user={
      Object {
        "first_name": "Jim",
        "id": "1234",
        "last_name": "Halpert",
        "mfa_active": "MFA",
        "nickname": "Big Tuna",
        "roles": "system_user",
        "username": "jim.halpert",
      }
    }
  />
  <ConfirmModal
    confirmButtonClass="btn btn-danger"
    confirmButtonText={
      <FormattedMessage
        defaultMessage="Deactivate"
        id="deactivate_member_modal.deactivate"
        values={Object {}}
      />
    }
    message={
      <div>
        <FormattedMessage
          defaultMessage="This action deactivates {username}. They will be logged out and not have access to any teams or channels on this system. Are you sure you want to deactivate {username}?"
          id="deactivate_member_modal.desc"
          values={
            Object {
              "username": "jim.halpert",
            }
          }
        />
        <strong>
          <br />
          <br />
          <FormattedMessage
            defaultMessage="You must also deactivate this user in the SSO provider or they will be reactivated on next login or sync."
            id="deactivate_member_modal.sso_warning"
            values={Object {}}
          />
        </strong>
      </div>
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <FormattedMessage
        defaultMessage="Deactivate {username}"
        id="deactivate_member_modal.title"
        values={
          Object {
            "username": "jim.halpert",
          }
        }
      />
    }
  />
</div>
`;

exports[`components/admin_console/system_user_detail should match snapshot if no nickname is defined 1`] = `
<div
  className="SystemUserDetail wrapper--fixed"
>
  <div
    className="admin-console__header with-back"
  >
    <div>
      <Connect(BlockableLink)
        className="fa fa-angle-left back"
        to="/admin_console/user_management/users"
      />
      <FormattedMessage
        defaultMessage="User Configuration"
        id="admin.systemUserDetail.title"
        values={Object {}}
      />
    </div>
  </div>
  <div
    className="admin-console__wrapper"
  >
    <div
      className="admin-console__content"
    >
      <AdminUserCard
        body={
          <React.Fragment>
            <span
              className="SystemUserDetail__position"
            />
            <span
              className="SystemUserDetail__field-label"
            >
              Email
            </span>
            <input
              className="SystemUserDetail__input form-control"
              onChange={[Function]}
              type="text"
            />
            <span
              className="SystemUserDetail__field-label"
            >
              Username
            </span>
            <p>
              jim.halpert
            </p>
            <span
              className="SystemUserDetail__field-label"
            >
              Authentication Method
            </span>
            <p>
              Email
            </p>
            <span
              className="SystemUserDetail__field-label"
            >
              Role
            </span>
            <p>
              <FormattedMessage
                defaultMessage="Member"
                id="admin.user_item.member"
                values={Object {}}
              />
            </p>
          </React.Fragment>
        }
        footer={
          <React.Fragment>
            <AdminButtonOutline
              className="admin-btn-default"
              onClick={[Function]}
            >
              Reset Password
            </AdminButtonOutline>
            <AdminButtonOutline
              className="admin-btn-default"
              onClick={[Function]}
            >
              Deactivate
            </AdminButtonOutline>
          </React.Fragment>
        }
        user={
          Object {
            "first_name": "Jim",
            "id": "1234",
            "last_name": "Halpert",
            "nickname": null,
            "roles": "system_user",
            "username": "jim.halpert",
          }
        }
      />
      <AdminPanel
        className=""
        subtitleDefault="Teams to which this user belongs"
        subtitleId="admin.userManagement.userDetail.teamsSubtitle"
        titleDefault="Team Membership"
        titleId="admin.userManagement.userDetail.teamsTitle"
      >
        <Connect(TeamList)
          userId="1234"
        />
      </AdminPanel>
    </div>
  </div>
  <div
    className="admin-console-save"
  >
    <SaveButton
      btnClass="btn-primary"
      disabled={true}
      extraClasses=""
      onClick={[Function]}
      saving={false}
      savingMessage="Saving Config..."
    />
    <div
      className="error-message"
    >
      <FormError
        error={null}
        errors={Array []}
      />
    </div>
    <Overlay
      animation={[Function]}
      delayShow={400}
      placement="top"
      rootClose={false}
      show={false}
    >
      <Tooltip
        bsClass="tooltip"
        id="error-tooltip"
        placement="right"
      />
    </Overlay>
  </div>
  <Connect(ResetPasswordModal)
    onModalDismissed={[Function]}
    onModalSubmit={[Function]}
    show={false}
    user={
      Object {
        "first_name": "Jim",
        "id": "1234",
        "last_name": "Halpert",
        "nickname": null,
        "roles": "system_user",
        "username": "jim.halpert",
      }
    }
  />
  <ConfirmModal
    confirmButtonClass="btn btn-danger"
    confirmButtonText={
      <FormattedMessage
        defaultMessage="Deactivate"
        id="deactivate_member_modal.deactivate"
        values={Object {}}
      />
    }
    message={
      <div>
        <FormattedMessage
          defaultMessage="This action deactivates {username}. They will be logged out and not have access to any teams or channels on this system. Are you sure you want to deactivate {username}?"
          id="deactivate_member_modal.desc"
          values={
            Object {
              "username": "jim.halpert",
            }
          }
        />
        <strong>
          <br />
          <br />
          <FormattedMessage
            defaultMessage="You must also deactivate this user in the SSO provider or they will be reactivated on next login or sync."
            id="deactivate_member_modal.sso_warning"
            values={Object {}}
          />
        </strong>
      </div>
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <FormattedMessage
        defaultMessage="Deactivate {username}"
        id="deactivate_member_modal.title"
        values={
          Object {
            "username": "jim.halpert",
          }
        }
      />
    }
  />
</div>
`;

exports[`components/admin_console/system_user_detail should match snapshot if user is inactive 1`] = `
<div
  className="SystemUserDetail wrapper--fixed"
>
  <div
    className="admin-console__header with-back"
  >
    <div>
      <Connect(BlockableLink)
        className="fa fa-angle-left back"
        to="/admin_console/user_management/users"
      />
      <FormattedMessage
        defaultMessage="User Configuration"
        id="admin.systemUserDetail.title"
        values={Object {}}
      />
    </div>
  </div>
  <div
    className="admin-console__wrapper"
  >
    <div
      className="admin-console__content"
    >
      <AdminUserCard
        body={
          <React.Fragment>
            <span
              className="SystemUserDetail__position"
            />
            <span
              className="SystemUserDetail__field-label"
            >
              Email
            </span>
            <input
              className="SystemUserDetail__input form-control"
              onChange={[Function]}
              type="text"
            />
            <span
              className="SystemUserDetail__field-label"
            >
              Username
            </span>
            <p>
              jim.halpert
            </p>
            <span
              className="SystemUserDetail__field-label"
            >
              Authentication Method
            </span>
            <p>
              Email
            </p>
            <span
              className="SystemUserDetail__field-label"
            >
              Role
            </span>
            <p>
              <FormattedMessage
                defaultMessage="Inactive"
                id="admin.user_item.inactive"
                values={Object {}}
              />
            </p>
          </React.Fragment>
        }
        footer={
          <React.Fragment>
            <AdminButtonOutline
              className="admin-btn-default"
              onClick={[Function]}
            >
              Reset Password
            </AdminButtonOutline>
            <AdminButtonOutline
              className="admin-btn-default"
              onClick={[Function]}
            >
              Activate
            </AdminButtonOutline>
          </React.Fragment>
        }
        user={
          Object {
            "delete_at": 1561683854166,
            "first_name": "Jim",
            "id": "1234",
            "last_name": "Halpert",
            "nickname": "Big Tuna",
            "roles": "system_user",
            "username": "jim.halpert",
          }
        }
      />
      <AdminPanel
        className=""
        subtitleDefault="Teams to which this user belongs"
        subtitleId="admin.userManagement.userDetail.teamsSubtitle"
        titleDefault="Team Membership"
        titleId="admin.userManagement.userDetail.teamsTitle"
      >
        <Connect(TeamList)
          userId="1234"
        />
      </AdminPanel>
    </div>
  </div>
  <div
    className="admin-console-save"
  >
    <SaveButton
      btnClass="btn-primary"
      disabled={true}
      extraClasses=""
      onClick={[Function]}
      saving={false}
      savingMessage="Saving Config..."
    />
    <div
      className="error-message"
    >
      <FormError
        error={null}
        errors={Array []}
      />
    </div>
    <Overlay
      animation={[Function]}
      delayShow={400}
      placement="top"
      rootClose={false}
      show={false}
    >
      <Tooltip
        bsClass="tooltip"
        id="error-tooltip"
        placement="right"
      />
    </Overlay>
  </div>
  <Connect(ResetPasswordModal)
    onModalDismissed={[Function]}
    onModalSubmit={[Function]}
    show={false}
    user={
      Object {
        "delete_at": 1561683854166,
        "first_name": "Jim",
        "id": "1234",
        "last_name": "Halpert",
        "nickname": "Big Tuna",
        "roles": "system_user",
        "username": "jim.halpert",
      }
    }
  />
  <ConfirmModal
    confirmButtonClass="btn btn-danger"
    confirmButtonText={
      <FormattedMessage
        defaultMessage="Deactivate"
        id="deactivate_member_modal.deactivate"
        values={Object {}}
      />
    }
    message={
      <div>
        <FormattedMessage
          defaultMessage="This action deactivates {username}. They will be logged out and not have access to any teams or channels on this system. Are you sure you want to deactivate {username}?"
          id="deactivate_member_modal.desc"
          values={
            Object {
              "username": "jim.halpert",
            }
          }
        />
        <strong>
          <br />
          <br />
          <FormattedMessage
            defaultMessage="You must also deactivate this user in the SSO provider or they will be reactivated on next login or sync."
            id="deactivate_member_modal.sso_warning"
            values={Object {}}
          />
        </strong>
      </div>
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <FormattedMessage
        defaultMessage="Deactivate {username}"
        id="deactivate_member_modal.title"
        values={
          Object {
            "username": "jim.halpert",
          }
        }
      />
    }
  />
</div>
`;

exports[`components/admin_console/system_user_detail should redirect if user id is not defined 1`] = `
<Redirect
  to={
    Object {
      "pathname": "/admin_console/user_management/users",
    }
  }
/>
`;
